http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/utils/NettyUtils.java

```diff
@@ -133,9 +133,11 @@ private static String getMessageForTooManyAcquireOperationsError() {
     }
 
     public static String closedChannelMessage(Channel channel) {
-        ChannelDiagnostics channelDiagnostics = channel.attr(CHANNEL_DIAGNOSTICS).get();
```

Comment:
```java

        ChannelDiagnostics channelDiagnostics = channel != null && channel.attr(CHANNEL_DIAGNOSTICS) != null ?
                                                channel.attr(CHANNEL_DIAGNOSTICS).get() : null;
        ChannelDiagnostics parentChannelDiagnostics = channel != null && channel.parent() != null ?
                                                      channel.parent().attr(CHANNEL_DIAGNOSTICS).get() : null;
```

Reply:
That looks good to me, I added a null check for the `CHANNEL_DIAGNOSTICS` attribute on the parent as well. Not sure if this passes style rules.